src/pydocstyle/utils.py

```diff
@@ -37,4 +39,10 @@ def common_prefix_length(a: str, b: str) -> int:
     for common, (ca, cb) in enumerate(zip(a, b)):
         if ca != cb:
             return common
-    return min(len(a), len(b))
\ No newline at end of file
+    return min(len(a), len(b))
+
+
+def strip_non_alphanumeric(string: str) -> str:
+    """Strip string from any non-alphanumeric characters."""
+    pattern = re.compile(r'[\W_]+')
```

Comment:
It doesn't make sense to do the inside the function definition. You won't benefit from doing a compile. I would move the compile outside, so that it's done only once.

Reply:
Done. Sorry for lon response time, I was on vacation.